public/app/plugins/datasource/grafana/datasource.ts

```diff
@@ -6,54 +6,63 @@ import {
   DataQueryResponse,
   DataSourceApi,
   DataSourceInstanceSettings,
+  LiveChannelScope,
 } from '@grafana/data';
 
-import { GrafanaQuery, GrafanaAnnotationQuery, GrafanaAnnotationType } from './types';
-import { getBackendSrv, getTemplateSrv, TemplateSrv, toDataQueryResponse } from '@grafana/runtime';
+import { GrafanaQuery, GrafanaAnnotationQuery, GrafanaAnnotationType, GrafanaQueryType } from './types';
+import { getBackendSrv, getTemplateSrv, toDataQueryResponse, getLiveMeasurmentsObserver } from '@grafana/runtime';
 import { Observable, of } from 'rxjs';
 import { map, catchError } from 'rxjs/operators';
 
+let counter = 100;
+
 export class GrafanaDatasource extends DataSourceApi<GrafanaQuery> {
-  constructor(
-    instanceSettings: DataSourceInstanceSettings,
-    private readonly templateSrv: TemplateSrv = getTemplateSrv()
-  ) {
+  constructor(instanceSettings: DataSourceInstanceSettings) {
     super(instanceSettings);
   }
 
   query(request: DataQueryRequest<GrafanaQuery>): Observable<DataQueryResponse> {
-    const { intervalMs, maxDataPoints, range, requestId } = request;
-
-    // Yes, this implementaiton ignores multiple targets!  But that matches exisitng behavior
-    const params: Record<string, any> = {
-      intervalMs,
-      maxDataPoints,
-      from: range.from.valueOf(),
-      to: range.to.valueOf(),
-    };
-
-    return getBackendSrv()
-      .fetch({
-        url: '/api/tsdb/testdata/random-walk',
-        method: 'GET',
-        params,
-        requestId,
-      })
-      .pipe(
-        map((rsp: any) => {
-          return toDataQueryResponse(rsp);
-        }),
-        catchError(err => {
-          return of(toDataQueryResponse(err));
-        })
-      );
+    const queries: Array<Observable<DataQueryResponse>> = [];
+    for (const target of request.targets) {
+      if (target.hide) {
+        continue;
+      }
+      if (target.queryType === GrafanaQueryType.LiveMeasurements) {
+        const { channel, measurements } = target;
+        if (channel) {
+          queries.push(
+            getLiveMeasurmentsObserver(
+              {
+                scope: LiveChannelScope.Grafana,
+                namespace: 'measurements',
+                path: channel,
+              },
+              `${request.requestId}.${counter++}`,
+              measurements
+            )
+          );
+        }
+      } else {
+        queries.push(getRandomWalk(request));
+      }
+    }
+    // With a single query just return the results
+    if (queries.length === 1) {
+      return queries[0];
+    }
+    if (queries.length > 1) {
+      // HELP!!!
+      return queries[0];
+    }
+    return of(); // nothing
```

Comment:
Maybe something like this?
```suggestion
     return forkJoin(queries).pipe(map((results:DataFrame[]) => ... )))
```

Reply:
I did something similar in https://github.com/grafana/grafana/pull/28221 so maybe we should brea this out into a utility function?